packages/builders/src/util/normalizeArray.ts

```diff
@@ -0,0 +1,6 @@
+export function normalizeArray<T>(...arr: T[] | T[][]): T[] {
+	if (Array.isArray(arr[0])) return [...arr[0]];
+	return [...arr] as T[];
+}
```

Comment:
```suggestion
export function normalizeArray<T>(arr: RestOrArray<T>): T[] {
	if (Array.isArray(arr[0])) return arr[0]
	return arr as T[];
}
```

and remove spreading from all places
```diff
- fields = normalizeArray(...fields);
+ fields = normalizeArray(fields);
```